checkout-core/src/main/java/com/adyen/checkout/core/AdyenLogger.kt

```diff
@@ -8,17 +8,75 @@
 
 package com.adyen.checkout.core
 
+import android.util.Log
+import com.adyen.checkout.core.internal.util.LogcatLogger
 import com.adyen.checkout.core.internal.util.Logger
 
 /**
  * Utility class to configure the Adyen logger.
  */
-object AdyenLogger {
+interface AdyenLogger {
 
-    /**
-     * Sets the minimum level to be logged.
-     */
-    fun setLogLevel(@Logger.LogLevel logLevel: Int) {
-        Logger.setLogLevel(logLevel)
+    fun shouldLog(level: AdyenLogLevel): Boolean
+
+    fun setLogLevel(level: AdyenLogLevel)
+
+    fun log(
+        level: AdyenLogLevel,
+        tag: String,
+        message: String,
+        throwable: Throwable?,
+    )
+
+    companion object {
+
+        @PublishedApi
+        @Volatile
+        internal var logger: AdyenLogger = LogcatLogger()
```

Comment:
When setting the logger, we delegate logging completely to that specific logger. I wonder if it would be beneficial to keep a list of loggers instead, to allow `LogcatLogger` and a custom logger(s) to handle logging simultaneously.
What do you think about this?

Reply:
I did take that into consideration, but decided having one logger would be better. Having multiple loggers could mean that the log will be logged multiple times, bloating the logs. Next to that, some merchants might not want to log to logcat for a good reason. Finally, logcat is not always working (take unit tests as example). Having a list with `LogcatLogger` always in there would mean it would crash if unavailable.